frontend/src/components/mine/Applications/MineApplicationInfo.js

```diff
@@ -0,0 +1,138 @@
+import {
+  fetchApplications,
+  updateApplication,
+  createApplication,
+} from "@/actionCreators/applicationActionCreator";
+
+/**
+ * @class  MinePermitInfo - contains all permit information
+ */
+
+const propTypes = {
+  mine: CustomPropTypes.mine.isRequired,
+  applications: PropTypes.arrayOf(CustomPropTypes.application),
+  openModal: PropTypes.func.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  fetchApplications: PropTypes.func.isRequired,
+  updateApplication: PropTypes.func.isRequired,
+  createApplication: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  applications: [],
+};
+
+export class MineApplicationInfo extends Component {
+  componentWillMount() {}
+
+  closeApplicationModal = () => {
+    this.props.closeModal();
+    this.props.fetchApplications({ mine_guid: this.props.mine.guid });
+  };
+
+  openAddApplicationModal = (event, title) => {
+    event.preventDefault();
+
+    this.props.openModal({
+      props: {
+        onSubmit: this.handleAddApplication,
+        title,
+      },
+      content: modalConfig.ADD_APPLICATION,
+    });
+  };
+
+  openEditApplicationModal = (event, application) => {
+    event.preventDefault();
+
+    const initialValues = { ...application };
+
+    this.props.openModal({
+      props: {
+        initialValues,
+        onSubmit: this.handleEditApplication,
+        title: `Edit application for ${application.application_no}`,
+      },
+      content: modalConfig.EDIT_APPLICATION,
+    });
+  };
+
+  handleAddApplication = (values) => {
+    const payload = { mine_guid: this.props.mine.guid, ...values };
+    return this.props.createApplication(payload).then(this.closeApplicationModal);
+  };
+
+  handleEditApplication = (values) =>
+    this.props.updateApplication(values.application_guid, values).then(this.closeApplicationModal);
+
+  render() {
+    return [
+      <div>
+        <div className="inline-flex between">
+          <div />
+          <div className="inline-flex between">
+            <AuthorizationWrapper inTesting>
+              <AuthorizationWrapper
+                permission={Permission.CREATE}
+                isMajorMine={this.props.mine.major_mine_ind}
+              >
+                <Button
+                  type="primary"
+                  onClick={(event) =>
+                    this.openAddApplicationModal(
+                      event,
+                      `${ModalContent.ADD_APPLICATION} to ${this.props.mine.mine_name}`
+                    )
+                  }
+                >
+                  <Icon type="plus" theme="outlined" style={{ fontSize: "18px" }} />
+                  Add a New Application
+                </Button>
+              </AuthorizationWrapper>
+            </AuthorizationWrapper>
+          </div>
+        </div>
+      </div>,
+      <br />,
+      <MineApplicationTable
+        applications={this.props.applications}
+        major_mine_ind={this.props.mine.major_mine_ind}
```

Comment:
should be in camelCase. 
`isMajorMine` is used in other places throughout the app for this boolean.

Reply:
Changed.